bigbio/hub/hub_repos/medparasimp/medparasimp.py

```diff
@@ -0,0 +1,250 @@
+                    "text_2_name": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "bigbio_t2t":
+            features = schemas.text2text_features
+        else:
+            raise ValueError(
+                f"Invalid config.schema specified ({self.config.schema}) - must be one of (source|bigbio_t2t)"
+            )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=str(_LICENSE),
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "doi_filepath": data_dir["train_doi"],
+                    "source_filepath": data_dir["train_source"],
+                    "target_filepath": data_dir["train_target"],
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "doi_filepath": data_dir["val_doi"],
+                    "source_filepath": data_dir["val_source"],
+                    "target_filepath": data_dir["val_target"],
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "doi_filepath": data_dir["test_doi"],
+                    "source_filepath": data_dir["test_source"],
+                    "target_filepath": data_dir["test_target"],
+                },
+            ),
+        ]
+
+    def _generate_examples(self, doi_filepath: str, source_filepath: str, target_filepath: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        # Read data from files
+        with open(doi_filepath, "r") as f:
+            dois: List[str] = f.read().splitlines()
+        with open(source_filepath, "r") as f:
+            sources: List[str] = f.read().splitlines()
+        with open(target_filepath, "r") as f:
+            targets: List[str] = f.read().splitlines()
+
+        if self.config.schema == "source":
+            for idx, (source, target) in enumerate(zip(sources, targets)):
+                key: int = idx
+                example: Dict = {
+                    "id": str(idx),
+                    "document_id": dois[idx],
+                    "text_1": source,
+                    "text_2": target,
+                    "text_1_name": "abstract",
+                    "text_2_name": "pls",
+                }
+                yield (key, example)
+        elif self.config.schema == "bigbio_t2t":
+            for idx, (source, target) in enumerate(zip(sources, targets)):
```

Comment:
Code for source and bigbio_t2t seems to be identical, can you merge these code blocks into one to avoid duplication?